core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -51,26 +51,35 @@ private[spark] class IndexShuffleBlockResolver(
 
   private val transportConf = SparkTransportConf.fromSparkConf(conf, "shuffle")
 
-  def getDataFile(shuffleId: Int, mapId: Int): File = {
-    blockManager.diskBlockManager.getFile(ShuffleDataBlockId(shuffleId, mapId, NOOP_REDUCE_ID))
+  def getDataFile(
+      shuffleId: Int,
+      mapId: Int,
+      shuffleGenerationId: Int = -1): File = {
```

Comment:
nit: move `shuffleGenerationId` to right after `shuffleId: Int`. We can add an overload version to implement parameter default value.

Reply:
Thanks, done in 96bab1e and remove the default value.